tensorflow/python/framework/dtypes.py

```diff
@@ -233,60 +233,135 @@ def __reduce__(self):
     np.float64: (-1, 1)
 }
 
+
+# Define classes for tensorflow dtypes
+class Float16(DType):
+  pass
+
+class Half(DType):
```

Comment:
For backward compatibility, this needs to be written like so: `Half = Float16`. Same for all other dtypes that are just the alias of another.